src/events.js

```diff
@@ -3192,172 +3150,116 @@ export const populateTrackingQuery = async (boxIdAndBagsObj) => {
+        const boxIdArray = Object.keys(boxIdAndBagsObj);
+        const trackingNumConfirmEls = Array.from(document.getElementsByClassName("invalid"));
+        if (trackingNumConfirmEls.length > 0) {
+          showNotifications({ title: 'Invalid Fields', body: 'Please add valid inputs to fields.' }, true);
+          return;
         }
 
-        for (let i = 0; i < boxIdArray.length; i++) {
-            let boxi = document.getElementById(boxIdArray[i] + "trackingId").value.toUpperCase();
-            let boxiConfirm = document.getElementById(boxIdArray[i] + "trackingIdConfirm").value.toUpperCase();
-            if (boxi == '' || boxiConfirm == '') {
-                emptyField = true
-                showNotifications({ title: 'Missing Fields', body: 'Please enter in shipment tracking numbers'}, true)
-                return
+        for (const boxId of boxIdArray) {
+            const trackingId = document.getElementById(boxId + "trackingId").value.toUpperCase();
+            const trackingIdConfirm = document.getElementById(boxId + "trackingIdConfirm").value.toUpperCase();
+            if (trackingId === '' || trackingIdConfirm === '') {
+                showNotifications({ title: 'Missing Fields', body: 'Please enter in shipment tracking numbers'}, true);
+                return;
             }
         
-            // if '959708259' exists update tracking number
-            if (boxIdAndBagsObj[boxIdArray[i]].hasOwnProperty('959708259')) {
-              boxIdAndBagsObj[boxIdArray[i]]['959708259'] = boxi
-            }
-            // if 'confirmTrackNum' exists update tracking number
-            if (boxIdAndBagsObj[boxIdArray[i]].hasOwnProperty('confirmTrackNum')) {
-              boxIdAndBagsObj[boxIdArray[i]]['confirmTrackNum'] = boxiConfirm 
-            }
-            // if specimens exists update, else add following key/values
-            if (boxIdAndBagsObj[boxIdArray[i]].hasOwnProperty('specimens')) {
-              boxIdAndBagsObj[boxIdArray[i]]['specimens'] = boxIdAndBagsObj[boxIdArray[i]]['specimens'] 
-            } 
-            else {
-              boxIdAndBagsObj[boxIdArray[i]] = { '959708259': boxi, confirmTrackNum: boxiConfirm, specimens: boxIdAndBagsObj[boxIdArray[i]] }
-            }  
+            shippingData.push({
+              [conceptIds.shippingTrackingNumber]: trackingId,
+              confirmTrackNum: trackingIdConfirm,
+              boxId
+            });
+            boxIdAndTrackingObj[boxId] = {
+              [conceptIds.shippingTrackingNumber]: trackingId,
+              specimens: boxIdAndBagsObj[boxId]
+            };
         }
 
-        let isDuplicateTrackingIdInDb = await checkDuplicateTrackingIdFromDb(boxIdArray );
-        
-        if(isDuplicateTrackingIdInDb || (checkFedexShipDuplicate(boxIdArray) && boxIdArray.length > 1)){
-            shippingDuplicateMessage()
-            return
+        const isDuplicateTrackingIdInDb = await checkDuplicateTrackingIdFromDb(boxIdArray );
```

Comment:
must get rid of space after `boxIdArray` !!! ;)

Reply:
Agreed!